packages/flutter_tools/test/general.shard/crash_reporting_test.dart

```diff
@@ -78,6 +81,37 @@ void main() {
       Stdio: () => const _NoStderr(),
     });
 
+    testUsingContext('should send only one crash report when async throws many', () async {
+      final Completer<int> exitCodeCompleter = Completer<int>();
+      setExitFunctionForTests((int exitCode) {
+        if (!exitCodeCompleter.isCompleted) {
+          exitCodeCompleter.complete(exitCode);
+        }
+      });
+
+      final RequestInfo requestInfo = RequestInfo();
+      final MockCrashReportSender sender = MockCrashReportSender(requestInfo);
+      CrashReportSender.initializeWith(sender);
+
+      FakeAsync().run((FakeAsync time) {
+        time.elapse(const Duration(seconds: 1));
+        unawaited(tools.run(
+          <String>['crash'],
+          <FlutterCommand>[_MultiCrashAsyncCommand(crashes: 4)],
+          reportCrashes: true,
+          flutterVersion: 'test-version',
+        ));
+        time.elapse(const Duration(seconds: 1));
+        time.flushMicrotasks();
+      });
+      expect(await exitCodeCompleter.future, equals(1));
```

Comment:
nit: I would use `expect(await foo, 1)`

Reply:
Done.

Reply:
I think it was the cause of the profile we are seeing from the frontend server crashers. I cycled through the error messages for a master version, and there was a discernible pattern.